apps/els_lsp/src/els_call_hierarchy_provider.erl

```diff
@@ -0,0 +1,112 @@
+-module(els_call_hierarchy_provider).
+
+-behaviour(els_provider).
+
+-export([ handle_request/2
+        , is_enabled/0
+        ]).
+
+%%==============================================================================
+%% Includes
+%%==============================================================================
+-include("els_lsp.hrl").
+-include_lib("kernel/include/logger.hrl").
+
+%%==============================================================================
+%% Defines
+%%==============================================================================
+
+%%==============================================================================
+%% Types
+%%==============================================================================
+-type state() :: any().
+
+%%==============================================================================
+%% els_provider functions
+%%==============================================================================
+-spec is_enabled() -> boolean().
+is_enabled() -> true.
+
+-spec handle_request(any(), state()) -> {any(), state()}.
+handle_request({prepare, Params}, State) ->
+  {Uri, Line, Char} =
+    els_text_document_position_params:uri_line_character(Params),
+  {ok, Document} = els_utils:lookup_document(Uri),
+  Functions = els_dt_document:wrapping_functions(Document, Line + 1, Char + 1),
+  Items = [function_to_item(Uri, F) ||  F <- Functions],
+  {Items, State};
+handle_request({incoming_calls, Params}, State) ->
+  #{ <<"item">> := #{<<"uri">> := Uri} = Item } = Params,
+  POI = els_call_hierarchy_item:poi(Item),
+  References = els_references_provider:find_references(Uri, POI),
+  Items = [reference_to_item(Reference) || Reference <- References],
+  {incoming_calls(Items), State};
+handle_request({outgoing_calls, Params}, State) ->
+  #{ <<"item">> := Item } = Params,
+  #{ <<"uri">> := Uri } = Item,
+  POI = els_call_hierarchy_item:poi(Item),
+  Applications = applications_in_function_range(Uri, POI),
```

Comment:
this took me a while to understand, applications are function applications, but in Erlang applications usually mean Erlang Applications.

Also I understand applications more as applications of the marked function. Maybe rename to calls_in_function_range or outgoing_calls

Reply:
Yeah, it can be confusing. This refers to `application` in [erl_syntax](https://erlang.org/doc/man/erl_syntax.html#application-2) terms. We should probably bring clarity around this, either by renaming or via a comment.